### PR TITLE
update dependencies for app

### PR DESCRIPTION
**Pull request by MarcoKorinth**

### CODE REVIEW

**Comment by MarcoKorinth** [comment]
The update of the i18next library caused an typescript error because one of its functions return type was changed from string to string|null. 
It's important to say that the function itself did not change, but only its type declaration.
In order to fix I had to add the file 'i18next.d.ts', which declares the module with custom type options.

Application functionality didn't change and all automatic and manual tests succeeded.

**Comment by MarcoKorinth** [comment]
As discussed earlier I'm going to replace the current i18next type issue solution with explicit null handling in the code.

File: LICENSE.md [hunk 1]
```diff
@@ -1025,48 +1025,45 @@ Section 6 -- Term and Termination.
...
+will not terminate this Public License.
 
+d. Sections 1, 5, 6, 7, and 8 survive termination of this Public
+License.
 
 Section 7 -- Other Terms and Conditions.
 
-  a. The Licensor shall not be bound by any additional or different
-     terms or conditions communicated by You unless expressly agreed.
-
```
Comment (tschmidtb51) [inline LICENSE.md:1041]: IMHO, this blank line should have been removed

File: app/lib/i18next/i18next.d.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import 'i18next'
+
+declare module 'i18next' {
+  interface CustomTypeOptions {
+    returnNull: false
+  }
+}
```
Comment (tschmidtb51) [inline app/lib/i18next/i18next.d.ts:7]: As discussed - implement an explicit check.

**Review by tschmidtb51 — CHANGES_REQUESTED** [review]
As discussed today.

Were there update available for bin/package-lock.json?

File: LICENSE.md (hunk 1 above)
Comment (MarcoKorinth) [reply LICENSE.md:1041]: I'm not sure what you mean.
There is a blank line between all subsections in every section, why remove it here?

**Comment by tschmidtb51** [comment]
> While looking for a solution for the possible null error problem it came to my attention that the t function only returns null if a null value was given in the translation.json file. We only use string values within this file and have no need for null values so this new feature / default behaviour is unnecessary for the secvisogram app. After consideration with @StefanNemeth we therefore propose to deactivate the accept null values feature in i18next and leave the customtypeoptions in the module declarations as proposed before. This procedure is also recommended in the official documentation: https://www.i18next.com/overview/typescript#argument-of-type-defaulttfuncreturn-is-not-assignable-to-parameter-of-type-xyz

Agree with the suggested solution. This is elegant.

File: app/lib/i18next/i18next.d.ts (hunk 2 above)
Comment (tschmidtb51) [reply app/lib/i18next/i18next.d.ts:7]: Fixed in [f1cf05e](https://github.com/secvisogram/secvisogram/pull/433/commits/f1cf05e13996aa5ec04656c6a448ef43e249c4da).

**Comment by tschmidtb51** [comment]
Just double checking that we do not miss anything: Does [f1cf05e](https://github.com/secvisogram/secvisogram/pull/433/commits/f1cf05e13996aa5ec04656c6a448ef43e249c4da) cover all relevant places?

File: LICENSE.md (hunk 1 above)
Comment (tschmidtb51) [reply LICENSE.md:1041]: Sorry - that was a reviewer mistake. My bad. Closing this as resolved.

**Review by StefanNemeth — APPROVED** [review]

**Comment by tschmidtb51** [comment]
Let me know, when it's ready for review...

**Comment by MarcoKorinth** [comment]
> Just double checking that we do not miss anything: Does [f1cf05e](https://github.com/secvisogram/secvisogram/pull/433/commits/f1cf05e13996aa5ec04656c6a448ef43e249c4da) cover all relevant places?

Yes it does.
According to the documentation the returnNull parameter in either customTypeOptions as well as in the i18next configuration options refer to the same thing.
So if there was another relevant place the tests would show a type error

**Review by tschmidtb51 — APPROVED** [review]
I didn't test it but from the code changes, it looks good to me

**Pull request merged**